python/deltalake/table.py

```diff
@@ -456,6 +456,27 @@ def vacuum(
             enforce_retention_duration,
         )
 
+    def update(
+        self,
+        updates: Dict[str, str],
+        predicate: Optional[str] = None,
+        writer_properties: Optional[Dict[str, int]] = None,
+        strict_cast: bool = True,
+    ) -> Dict[str, Any]:
+        """UPDATE records in the Delta Table that matches an optional predicate.
+
+        :param updates: what values to update
+        :param predicate: a logical expression, defaults to None
+        :writer_properties: Pass writer properties to the Rust parquet writer, see options https://arrow.apache.org/rust/parquet/file/properties/struct.WriterProperties.html
+        :strict_cast: specify if data types need to be casted strictly or not :default = True
+        :return: the metrics from delete
+        """
+
+        metrics = self._table.update(
+            updates, predicate, writer_properties, safe_cast=not strict_cast
```

Comment:
It doesn't seem right to me that `safe_cast` and `strict_cast` would be antonyms. Do we have a test validating this?

Reply:
I will add the test. Reason why I named it strict_cast is that I found safe_cast in rust confusing. Safe_cast = False means it will give an error when the types are not the same.